### DISCUSSION TITLE
Be able to set a default color for Custom Overlays

### DISCUSSION THREAD

**Discussion by ravenfeld** [Helium314/SCEE#560] [Ideas]
I notice that you can't set a default colour, it's always LIME.

Have you had any requests for this? I can try and see if I can choose a color. For the moment I've modified `if (colorKeySelector == null) Color.LIME` in my code directly

**Comment by mnalis** [comment]
> I notice that you can't set a default colour, it's always LIME.

When **creating new custom overlay**, on the top-left corner of the _"Custom overlay"_ window you will see lime circle (with  triangle indicating [drop-down list ](https://en.wikipedia.org/wiki/Drop-down_list) right of it). If you **click on that lime circle**, you can any other icon, or (at a bottom of the scrollable list) a different colored circles.

Does that work for you? Or did I misunderstood the issue?

**Reply by ravenfeld** [reply]
Yes, it works, but it doesn't change the color on the map. 

![Screenshot_20240701_172925](https://github.com/Helium314/SCEE/assets/3765114/bab33912-0c86-42e5-9f47-f557a5893b9d)

But maybe that's just how I use it, I like to be able to select ways along their entire length and not just on the pine. So I often use the surface overlay for that, but with custom I've found the same thing, only better.

**Reply by mnalis** [reply]
Ah, I see, you didn't mean color the overlay icon, but instead what colors is used for coloring ways/nodes on the map that match the element selection criteria.

I'm not sure however what you meant by _"not just on the pine"_ ? I think that in both Surface overlay ***and*** in custom overlay clicking on the way should select the exactly the same element?

---

As for the color used to color ways/POIs on the map, in the custom overlay there is _"Coloring details"_ options where you can choose how they will be colored. If it is empty, they'll all be lime color; but if specify some other selection method, the color will change automatically (e.g. if you type `surface` there, every different value of `surface` tag will get different color). But you cannot manually select color there AFAIK.

**Reply by ravenfeld** [reply]
I didn't understand how to use it at the time.
Because I want to display `way with highway=*` with a colour of my choice
